src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/PersonName.java

```diff
@@ -0,0 +1,62 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.EdifactValidationException;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Builder
+@Data
+public class PersonName extends Segment {
+
+    private @NonNull String nhsNumber;
+    private @NonNull String surname;
+    private String forename;
+    private String title;
+    private String middleName;
+    private String[] otherNames;
+
+    @Override
+    public String getKey() {
+        return "PNA";
+    }
+
+    @Override
+    public String getValue() {
```

Comment:
This implementation will leave `+` at the end of Value and we should trim them.
I made it this way:
````java
 @Override
    public String getValue() {
        List<String> values = new ArrayList<>();

        String namesDelimiter = containsName() ? "++" : "";
        Optional.ofNullable(this.nhsNumber)
            .map(value -> value + ":" + this.patientIdentificationType + namesDelimiter)
            .ifPresentOrElse(values::add, () -> values.add(namesDelimiter));

        Optional.ofNullable(this.familyName)
            .map(value -> "SU:" + value)
            .ifPresent(values::add);
        Optional.ofNullable(this.forename)
            .map(value -> "FO:" + value)
            .ifPresent(values::add);
        Optional.ofNullable(this.title)
            .map(value -> "TI:" + value)
            .ifPresent(values::add);
        Optional.ofNullable(this.middleName)
            .map(value -> "MI:" + value)
            .ifPresent(values::add);
        Optional.ofNullable(this.thirdForename)
            .map(value -> "FS:" + value)
            .ifPresent(values::add);

        return QUALIFIER + "+" + values.stream().collect(Collectors.joining("+"));
    }
````

Reply:
or just trim unnecessary "+" from the end